File: BRaVe-Management-Backend/Models/SurveyQuestion.cs
using Microsoft.AspNetCore.Http.HttpResults;
using System;

namespace BRaVe_Management_Backend.Models
{

    public class SurveyQuestion
    {
        public string SurveyCode { get; set; }
        public int Id { get; set; }
        public int TenantId { get; set; }
        public int QuestionOrder { get; set; }
        public string DefaultLang { get; set; }
        public bool IsRequired { get; set; }
        public int AnswerType { get; set; }
        public int? LookupId { get; set; }
        public int? DatasetId { get; set; }
        public short? MinSelection { get; set; }
        public short? MaxSelection { get; set; }
        public string? Restriction { get; set; }
        public string? SkipLogic { get; set; }
        public string? ResultExpression { get; set; }
        public bool IsActive { get; set; }
        public string CreatedByUserId { get; set; }
        public DateTime CreatedOn { get; set; }
        public string? UpdatedByUserId { get; set; }
        public DateTime? UpdatedOn { get; set; }
    }
}

File: BRaVeMobile/app/src/main/java/intl/iom/bravemobile/models/datapoints/SurveyQuestion.java
package intl.iom.bravemobile.models.datapoints;

import java.util.List;

import intl.iom.bravemobile.models.activities.QuestionModel;
import intl.iom.bravemobile.statics.AnswerType;

public class SurveyQuestion extends CollectionUnit
{
    public boolean isActive;
    public SurveyQuestion(int id, boolean isActive, int order, AnswerType answerType, Integer lookupId, Integer datasetId, boolean is_required, Integer minSelection, Integer maxSelection, String skipLogic, String restriction, String resultEvaluation, List<TranslatedText> translations) {
        super(id, order, answerType, lookupId, datasetId, is_required, minSelection, maxSelection, skipLogic, restriction, resultEvaluation, translations);
        this.isActive = isActive;
    }


    public SurveyQuestion(int id, boolean isActive, int order, AnswerType answerType, Integer lookupId, Integer datasetId, boolean is_required, Integer minSelection, Integer maxSelection, String skipLogic, String restriction, String resultEvaluation, String text) {
        super(id, order, answerType, lookupId, datasetId, is_required, minSelection, maxSelection, skipLogic, restriction, resultEvaluation, text);
        this.isActive = isActive;
    }

    public SurveyQuestion(QuestionModel restored) {
        super(restored.id, restored.order, AnswerType.fromCode(restored.type), restored.lookup, restored.dataset, restored.isRequired, restored.minVal, restored.maxVal, restored.skipLogic,  restored.restriction, restored.resultExpression, restored.texts);
        this.isActive = true; //will remove later
    }
}
